pkg/tokenization/pool.go

```diff
@@ -79,16 +87,29 @@ func NewTokenizationPool(config *Config, store prefixstore.Indexer) (*Pool, erro
 	}, nil
 }
 
-// AddTask enqueues a new tokenization task.
+// EnqueueTokenization enqueues a new tokenization task.
 // This method only enqueues the task and does not start processing it.
-func (pool *Pool) AddTask(prompt, modelName string) {
+func (pool *Pool) EnqueueTokenization(prompt, modelName string) {
 	task := Task{
 		Prompt:    prompt,
 		ModelName: modelName,
 	}
 	pool.queue.Add(task)
 }
 
+// Tokenize queues a task and blocks until the final result is available.
+func (pool *Pool) Tokenize(prompt, modelName string) TokenizationResponse {
```

Comment:
Since the offsets use is completely internal - I think this function can just return `[]uint32` (token IDs), and the `TokenizationResponse` type can be hidden.